File: Data Structures & Algorithms/binary-search/submission-6.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)-1

        while l <= r:
            mid = (r + l) // 2
            val = nums[mid]
            if target < val:
                r = mid - 1
            elif target > val:
                l = mid + 1
            else:
                return mid
        return -1
        

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-6.go
func minSubArrayLen(target int, nums []int) int {
    minLen := len(nums) + 1
    l := 0
    total := 0

    for r := 0; r < len(nums); r++ {
        total += nums[r]
        for total >= target {
            minLen = min(minLen, r - l + 1)
            total -= nums[l]
            l++
        }

    }
    if minLen == len(nums) + 1 {
        return 0
    }
    return minLen
}
